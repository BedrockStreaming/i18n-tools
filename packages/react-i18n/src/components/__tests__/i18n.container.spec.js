import React from 'react';
import { shallow } from 'enzyme';
import { translate } from '../i18n.container';

jest.unmock('../i18n.container');

describe('i18n container', () => {
  let context;
  let listeners;
  const DummyComponent = props => <div {...props} />;
  const TranslatedComponent = translate(DummyComponent);

  beforeEach(() => {
    listeners = [];
    context = {
      getTranslateFunction: jest.fn(() => jest.fn()),
    };
  });

  it('should provide translate function and inherited props', () => {
    const wrapper = shallow(<TranslatedComponent foo="bar" />, { context });

    expect(wrapper).toMatchSnapshot();
  });
});
