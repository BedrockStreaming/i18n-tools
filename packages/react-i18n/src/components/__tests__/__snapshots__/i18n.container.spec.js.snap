// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`i18n container should provide translate function and inherited props 1`] = `
<DummyComponent
  foo="bar"
  t={[Function]}
/>
`;
